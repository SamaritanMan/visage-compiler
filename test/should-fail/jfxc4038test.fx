/*
* Regression test:  An exception has occurred in the OpenJavafx compiler
*
* @compilearg -sourcepath
* @compilearg test/should-fail
* @test/compile-error
*/
public class jfxc4038test {
var test:jfxc4038sub;
public function func() {
test.f;
1
}
}
