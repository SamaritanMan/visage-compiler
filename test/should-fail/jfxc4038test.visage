/*
* Regression test:  An exception has occurred in the Visage compiler
*
* @compilearg -sourcepath
* @compilearg test/should-fail
* @test/compile-error
*/
public class jfxc4038test {
var test:jfxc4038sub;
public function func() {
test.f;
1
}
}
