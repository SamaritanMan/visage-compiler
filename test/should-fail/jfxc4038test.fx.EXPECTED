test/should-fail/jfxc4038test.fx:10: Note: Need explicit type for func because of cycle in definition.
public function func() {
^
test/should-fail/jfxc4038sub.fx:6: This is the cyclic reference to func that prevents type inference.
public var f=function(){jfxc4038test{}.func()}
                                      ^
1 error
